Add Support for classes defined with compact style

sinsoku opened the pull request:
There is a problem that `Rails/UniqueValidationWithoutIndex` doesn't work
because it can't find the table name in the class defined with compact style.

This commit adds test code to ensure existing behavior and adds support for
classes defined with compact style.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop-rails/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop-rails/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.
* [ ] If this is a new cop, consider making a corresponding update to the [Rails Style Guide](https://github.com/rubocop-hq/rails-style-guide).

[1]: https://chris.beams.io/posts/git-commit/

koic commented:
Can you rebase with the latest master branch and add a reproduction test case to spec/rubocop/cop/rails/unique_validation_without_index_spec.rb?

sinsoku commented:
I rebased and force-pushed.

The test case of "when the class is defined with compact style" is the reproduction test case.
Should I split commits?

koic commented:
I think it's happening the user level issue at `Rails/UniqueValidationWithoutIndex` cop. spec/rubocop/cop/active_record_helper_spec.rb has written for `ActiveRecordHelper` internal API test case.
I'm hoping that a test for `Rails/UniqueValidationWithoutIndex` cop will be added to spec/rubocop/cop/rails/unique_validation_without_index_spec.rb.

sinsoku commented:
Thanks for your review comment.
I understood.

I added a test case for `Rails/UniqueValidationWithoutIndex` and force-pushed it.

koic commented:
Nice 👍 Great thanks!
